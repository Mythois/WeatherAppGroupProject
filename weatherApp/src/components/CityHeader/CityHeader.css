.cityHeader {
    background-color: var(--cmpbg);
    border-radius: 20px;
    width: 90%;
    height: 20%;
    padding: 1rem;
}