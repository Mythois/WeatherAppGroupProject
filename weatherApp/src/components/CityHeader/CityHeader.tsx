import React from 'react'
import './CityHeader.css'

import getWeatherIcon from '../../utils/getWeatherIcon'

type CityHeaderProps = {
    cityName: string,
    cityTempMax: number;
    cityTempMin: number;
    cityPersipitation: number;
    cloudCoverage: number;
}

function CityHeader({cityName, cityTempMax, cityTempMin, cityPersipitation, cloudCoverage}: CityHeaderProps) {
    return (
        <div className='cityHeader'>
            <h4>{cityName}</h4>
            <img src={getWeatherIcon(cityPersipitation, cloudCoverage)} alt="icon" />
            <p>{cityTempMax} / {cityTempMin}â„ƒ</p>
            <p>{cityPersipitation}mm</p>
        </div>
        )
}

export default CityHeader
