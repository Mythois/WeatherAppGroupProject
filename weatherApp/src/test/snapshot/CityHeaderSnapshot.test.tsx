import { render } from '@testing-library/react'
import CityHeader from '../../components/CityHeader/CityHeader'

test('CityHeader snapshot test', () => {
  const props = {
    cityName: 'Example City',
    cityTempMax: 30,
    cityTempMin: 20,
    cityPrecipitation: 10,
    cloudCoverage: 50,
  }

  const { container } = render(<CityHeader {...props} />)
  expect(container).toMatchSnapshot('src/test/snapshot/__snapshots__/CityHeaderSnapshot.test.tsx.snap')
})
