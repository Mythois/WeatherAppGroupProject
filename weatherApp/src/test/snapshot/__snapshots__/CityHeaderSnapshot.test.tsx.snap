// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CityHeader snapshot test > src	estsnapshot__snapshots__CityHeaderSnapshot.test.tsx.snap 1`] = `
<div>
  <div
    class="cityHeader"
  >
    <h3>
      Example City
    </h3>
    <img
      alt="icon"
      src="/src/assets/rain.svg"
    />
    <p>
      10
       mm
    </p>
    <p>
      30
       / 
      20
      ℃
    </p>
  </div>
</div>
`;

exports[`CityHeader snapshot test > src/test/snapshot/__snapshots__/CityHeaderSnapshot.test.tsx.snap 1`] = `
<div>
  <div
    class="cityHeader"
  >
    <h3>
      Example City
    </h3>
    <img
      alt="icon"
      src="/src/assets/rain.svg"
    />
    <p>
      10
       mm
    </p>
    <p>
      30
       / 
      20
      ℃
    </p>
  </div>
</div>
`;

exports[`CityHeader snapshot test > src/test/snapshot/__snapshots__CityHeaderSnapshot.test.tsx.snap 1`] = `
<div>
  <div
    class="cityHeader"
  >
    <h3>
      Example City
    </h3>
    <img
      alt="icon"
      src="/src/assets/rain.svg"
    />
    <p>
      10
       mm
    </p>
    <p>
      30
       / 
      20
      ℃
    </p>
  </div>
</div>
`;

exports[`CityHeader snapshot test > src/test/snapshot__snapshots__CityHeaderSnapshot.test.tsx.snap 1`] = `
<div>
  <div
    class="cityHeader"
  >
    <h3>
      Example City
    </h3>
    <img
      alt="icon"
      src="/src/assets/rain.svg"
    />
    <p>
      10
       mm
    </p>
    <p>
      30
       / 
      20
      ℃
    </p>
  </div>
</div>
`;

exports[`CityHeader snapshot test > src/testsnapshot__snapshots__CityHeaderSnapshot.test.tsx.snap 1`] = `
<div>
  <div
    class="cityHeader"
  >
    <h3>
      Example City
    </h3>
    <img
      alt="icon"
      src="/src/assets/rain.svg"
    />
    <p>
      10
       mm
    </p>
    <p>
      30
       / 
      20
      ℃
    </p>
  </div>
</div>
`;

exports[`CityHeader snapshot test 1`] = `
<div>
  <div
    class="cityHeader"
  >
    <h3>
      Example City
    </h3>
    <img
      alt="icon"
      src="/src/assets/rain.svg"
    />
    <p>
      10
       mm
    </p>
    <p>
      30
       / 
      20
      ℃
    </p>
  </div>
</div>
`;
